### ISSUE TITLE
Dragging a ROI file to Fiji cause an exception

### ISSUE THREAD

**Issue by hadim** [imagej/imagej-legacy#148]
It happens with both .zip and .roi files. The error occurs (SCIFIO related) but the ROI file still opens in the ROIManager. I guess there is an IJ1 fallback mechanism that finally finds out how to open the file.

Would it be possible to catch/hide the error ? Because each time I open a ROI file manually a window popup and it's pretty annoying.

**Comment by hadim** [comment]
The error is : 

```
java.io.IOException: io.scif.img.ImgIOException: java.io.IOException: io.scif.FormatException: 0000-0059.roi: No supported format found.
	at io.scif.services.DefaultDatasetIOService.open(DefaultDatasetIOService.java:136)
	at io.scif.services.DefaultDatasetIOService.open(DefaultDatasetIOService.java:111)
	at io.scif.io.DatasetIOPlugin.open(DatasetIOPlugin.java:75)
	at io.scif.io.DatasetIOPlugin.open(DatasetIOPlugin.java:50)
	at net.imagej.legacy.plugin.DefaultLegacyOpener.open(DefaultLegacyOpener.java:135)
	at net.imagej.legacy.DefaultLegacyHooks.interceptDragAndDropFile(DefaultLegacyHooks.java:363)
	at ij.plugin.DragAndDrop.openFile(DragAndDrop.java)
	at ij.plugin.DragAndDrop.run(DragAndDrop.java:159)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.scif.img.ImgIOException: java.io.IOException: io.scif.FormatException: 0000-0059.roi: No supported format found.
	at io.scif.img.ImgOpener.createReader(ImgOpener.java:571)
	at io.scif.img.ImgOpener.openImgs(ImgOpener.java:145)
	at io.scif.services.DefaultDatasetIOService.open(DefaultDatasetIOService.java:126)
	... 8 more
Caused by: java.io.IOException: io.scif.FormatException: 0000-0059.roi: No supported format found.
	at io.scif.AbstractReader.setSource(AbstractReader.java:274)
	at io.scif.services.DefaultInitializeService.initializeReader(DefaultInitializeService.java:90)
	at io.scif.img.ImgOpener.createReader(ImgOpener.java:562)
	... 10 more
Caused by: io.scif.FormatException: 0000-0059.roi: No supported format found.
	at io.scif.services.DefaultFormatService.getFormatList(DefaultFormatService.java:351)
	at io.scif.services.DefaultFormatService.getFormat(DefaultFormatService.java:317)
	at io.scif.formats.ZipFormat$Parser.typedParse(ZipFormat.java:174)
	at io.scif.formats.ZipFormat$Parser.typedParse(ZipFormat.java:148)
	at io.scif.AbstractParser.parse(AbstractParser.java:253)
	at io.scif.formats.ZipFormat$Parser.parse(ZipFormat.java:161)
	at io.scif.formats.ZipFormat$Parser.parse(ZipFormat.java:148)
	at io.scif.AbstractParser.parse(AbstractParser.java:336)
	at io.scif.AbstractParser.parse(AbstractParser.java:52)
	at io.scif.AbstractReader.setSource(AbstractReader.java:270)
	... 12 more
```

**Comment by hadim** [comment]
Instead of printing the traceback, IJ could just inform the user that IJ1 has been used to open this file (because SCIFIO couldn't).

**Comment by rimadoma** [comment]
Hmm, I could try to write "null object" implementations to SCIFIO for .zip and .roi that recognise the formats, don't do anything with them. This might stop the fall through to ImageJ1 one though. Let me see what happens.

**Comment by hadim** [comment]
Thank you @rimadoma . Let's see if the null object do the trick.

**Comment by rimadoma** [comment]
A quick update: there's [an implementation](https://github.com/scifio/scifio/blob/master/src/main/java/io/scif/formats/ZipFormat.java) for zip-files in SCIFIO. It worked for me when, for example, I opened a package of DICOM images. However, I did get an exception if I tried to open an encrypted zip-file, or a package with no images in it. 

TL;DR At the moment there's support for zipped images, but no general support for zip-files

**Comment by rimadoma** [comment]
Okay so I tried writing a simple [null object implementation](https://github.com/bonej-org/scifio/tree/null-formats) for the ROI format, and what happened is what I suspected: since SCIFIO finds an implementation for the file, it doesn't "fall through" to ImageJ1 any more. SCIFIO asks the class to open the file, and since it doesn't do anything, an empty image pops open. To put it shortly either you don't have an implementation or implement it properly.

Imho this issue opens a point for discussion: I don't think unsupported file format should throw an expression. It's not that uncommon.
